test/extensions/filters/http/oauth2/config_test.cc

```diff
@@ -22,33 +22,101 @@ namespace Oauth2 {
 using testing::NiceMock;
 using testing::Return;
 
+namespace {
+
+// This loads one of the secrets in credentials, and fails the other one.
+void expectInvalidSecretConfig(const std::string& correctly_loaded_secret,
+                               const std::string& exception_message) {
+  const std::string yaml = R"EOF(
+config:
+  token_endpoint:
+    cluster: foo
+    uri: oauth.com/token
+    timeout: 3s
+  credentials:
+    client_id: "secret"
+    token_secret:
+      name: token
+    hmac_secret:
+      name: hmac
+  authorization_endpoint: https://oauth.com/oauth/authorize/
+  redirect_uri: "%REQ(:x-forwarded-proto)%://%REQ(:authority)%/callback"
+  redirect_path_matcher:
+    path:
+      exact: /callback
+  signout_path:
+    path:
+      exact: /signout
+    )EOF";
+
+  OAuth2Config factory;
+  ProtobufTypes::MessagePtr proto_config = factory.createEmptyConfigProto();
+  TestUtility::loadFromYaml(yaml, *proto_config);
+  NiceMock<Server::Configuration::MockFactoryContext> context;
+
+  auto& secret_manager = context.cluster_manager_.cluster_manager_factory_.secretManager();
+  ON_CALL(secret_manager, findStaticGenericSecretProvider(correctly_loaded_secret))
+      .WillByDefault(Return(std::make_shared<Secret::GenericSecretConfigProviderImpl>(
+          envoy::extensions::transport_sockets::tls::v3::GenericSecret())));
+
+  EXPECT_THROW_WITH_MESSAGE(factory.createFilterFactoryFromProto(*proto_config, "stats", context),
+                            EnvoyException, exception_message);
+}
+
+} // namespace
+
 TEST(ConfigTest, CreateFilter) {
   const std::string yaml = R"EOF(
 config:
-    token_endpoint:
-      cluster: foo
-      uri: oauth.com/token
-      timeout: 3s
-    authorization_endpoint: https://oauth.com/oauth/authorize/
-    redirect_uri: "%REQ(:x-forwarded-proto)%://%REQ(:authority)%/callback"
-    signout_path: 
-      path:
-        exact: /signout
+  token_endpoint:
+    cluster: foo
+    uri: oauth.com/token
+    timeout: 3s
+  credentials:
+    client_id: "secret"
+    token_secret: {}
+    hmac_secret: {}
+  authorization_endpoint: https://oauth.com/oauth/authorize/
+  redirect_uri: "%REQ(:x-forwarded-proto)%://%REQ(:authority)%/callback"
+  redirect_path_matcher:
+    path:
+      exact: /callback
+  signout_path:
+    path:
+      exact: /signout
     )EOF";
 
-  envoy::extensions::filters::http::oauth2::v3alpha::OAuth2 proto_config;
-  MessageUtil::loadFromYaml(yaml, proto_config, ProtobufMessage::getStrictValidationVisitor());
-  NiceMock<Server::Configuration::MockFactoryContext> factory_context;
-  auto& secret_manager = factory_context.cluster_manager_.cluster_manager_factory_.secretManager();
+  OAuth2Config factory;
+  ProtobufTypes::MessagePtr proto_config = factory.createEmptyConfigProto();
+  TestUtility::loadFromYaml(yaml, *proto_config);
+  testing::StrictMock<Server::Configuration::MockFactoryContext> context;
```

Comment:
I think the mocks are strict by default? Does specifying `StrictMock` do anything?

Reply:
Removed. Seems like I need to clean up the ext_authz one.